File: BrasilBurger/Models/MenuBurger.cs
namespace Models
{
    using System.ComponentModel.DataAnnotations.Schema;
    public class MenuBurger
    {
        [Column("id")]
        public long Id { get; set; }
        [Column("menu_id")]
        public long MenuId { get; set; }
        [Column("burger_id")]
        public long BurgerId { get; set; }
        [Column("quantite")]
        public int Quantite { get; set; }

        [ForeignKey("BurgerId")]
        public Burger? Burger { get; set; }
        [ForeignKey("MenuId")]
        public Menu? Menu { get; set; }
    }

}

File: src/Entity/MenuBurger.php
<?php

namespace App\Entity;

use App\Repository\MenuBurgerRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MenuBurgerRepository::class)]
#[ORM\Table(name: 'menu_burger')]
class MenuBurger
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'menuBurgers')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Menu $menu = null;

    #[ORM\ManyToOne(inversedBy: 'menuBurgers')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Burger $burger = null;

    #[ORM\Column]
    private ?int $quantite = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(string $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getMenu(): ?Menu
    {
        return $this->menu;
    }

    public function setMenu(?Menu $menu): static
    {
        $this->menu = $menu;

        return $this;
    }

    public function getBurger(): ?Burger
    {
        return $this->burger;
    }

    public function setBurger(?Burger $burger): static
    {
        $this->burger = $burger;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): static
    {
        $this->quantite = $quantite;

        return $this;
    }
}
